activesupport/test/error_reporter_test.rb

```diff
@@ -166,9 +166,21 @@ class ErrorReporterTest < ActiveSupport::TestCase
     assert_nil error.backtrace
     assert_nil error.backtrace_locations
 
-    assert_nil @reporter.report(error)
-    assert_not_predicate error.backtrace, :empty?
-    assert_not_predicate error.backtrace_locations, :empty?
```

Comment:
These tests were too lax, and don't cover the backtrace shifting logic.

It's wrong when running in Spring, but that's a Spring bug (https://github.com/rails/spring/issues/739)